pkgs/misc/emulators/citra/default.nix

```diff
@@ -1,91 +1,30 @@
+    pname = "citra-${branch}";
+    version = "1742";
+    branchName = branch;
+    branchDesc = "A Nintendo 3DS Emulator, this branch contains already reviewed and tested features";
+    src = fetchFromGitHub {
+      owner = "citra-emu";
+      repo = pname;
+      rev = "${branch}-${version}";
+      sha256 = "4sdiDmqvOP8GH8luxhikS8N0DR/AEC1dgcA5pCzlPsw=";
+      fetchSubmodules = true;
+    };
   };
-
-  nativeBuildInputs = [
-    cmake
-    pkg-config
-  ]
-  ++ lib.optionals enableQt [ wrapQtAppsHook ];
-
-  buildInputs = [
-    boost17x
-    libusb1
-  ]
-  ++ lib.optionals enableSdl2 [ SDL2 ]
-  ++ lib.optionals enableQt [ qtbase qtmultimedia ]
-  ++ lib.optionals enableQtTranslation [ qttools ]
-  ++ lib.optionals enableCubeb [ libpulseaudio ]
-  ++ lib.optionals (enableFfmpegAudioDecoder || enableFfmpegVideoDumper) [ ffmpeg ]
-  ++ lib.optionals useDiscordRichPresence [ rapidjson ]
-  ++ lib.optionals enableFdk [ fdk_aac ];
-
-  cmakeFlags = [
-    "-DUSE_SYSTEM_BOOST=ON"
-  ]
-  ++ lib.optionals (!enableSdl2) [ "-DENABLE_SDL2=OFF" ]
-  ++ lib.optionals (!enableQt) [ "-DENABLE_QT=OFF" ]
-  ++ lib.optionals enableQtTranslation [ "-DENABLE_QT_TRANSLATION=ON" ]
-  ++ lib.optionals (!enableWebService) [ "-DENABLE_WEB_SERVICE=OFF" ]
-  ++ lib.optionals (!enableCubeb) [ "-DENABLE_CUBEB=OFF" ]
-  ++ lib.optionals enableFfmpegAudioDecoder [ "-DENABLE_FFMPEG_AUDIO_DECODER=ON"]
-  ++ lib.optionals enableFfmpegVideoDumper [ "-DENABLE_FFMPEG_VIDEO_DUMPER=ON" ]
-  ++ lib.optionals useDiscordRichPresence [ "-DUSE_DISCORD_PRESENCE=ON" ]
-  ++ lib.optionals enableFdk [ "-DENABLE_FDK=ON" ];
-
-  postPatch = ''
-    # We already know the submodules are present
-    substituteInPlace CMakeLists.txt \
-      --replace "check_submodules_present()" ""
-
-    # Devendoring
-    rm -rf externals/zstd externals/libressl
-    cp -r ${zstd.src} externals/zstd
-    tar xf ${libressl.src} -C externals/
-    mv externals/${libressl.name} externals/libressl
-    chmod -R a+w externals/zstd
-  '';
-
-  # Todo: cubeb audio backend (the default one) doesn't work on the SDL interface.
-  # This seems to be a problem with libpulseaudio, other applications have similar problems (e.g Duckstation).
-  # Note that the two interfaces have two separate configuration files.
-
-  meta = with lib; {
-    homepage = "https://citra-emu.org";
-    description = "An open-source emulator for the Nintendo 3DS";
-    license = licenses.gpl2;
-    maintainers = with maintainers; [ abbradar ];
-    platforms = platforms.linux;
+  canary = libsForQt5.callPackage ./base.nix rec {
+    pname = "citra-${branch}";
+    version = "2019";
+    branchName = branch;
+    branchDesc = "A Nintendo 3DS Emulator, this branch contains additional features still under review";
+    src = fetchFromGitHub {
+      owner = "citra-emu";
+      repo = pname;
+      rev = "${branch}-${version}";
+      sha256 = "btHcctjeoibg6lbAv8LZ9h/vjJnzeWCgiYosnNUJkUs=";
+      fetchSubmodules = true;
```

Comment:
We should use the system zstd, libressl (the changes done in the fork do not seem code related),

Reply:
zstd is currently statically linked, I can replace the zstd sources from Nix or we can try to patch the build system but I'm not a cmake expert.

libressl is forked as you mentioned, but also built with 'HAVE_INET_NTOP' - not sure what this means but the derivation in nixpkgs is not built with this option.